leshan-server-core/src/main/java/org/eclipse/leshan/server/send/SendHandler.java

```diff
@@ -90,10 +91,11 @@ public void run() {
                 if (sendResponse.isSuccess()) {
                     fireDataReceived(updatedRegistration, request.getTimestampedNodes(), request);
                 } else {
-                    onError(updatedRegistration, String.format("Invalid Send Request, server returns %s %s", //
+                    String errorMsg = String.format("Invalid Send Request, server returns %s %s", //
                             sendResponse.getCode().getName(), //
-                            sendResponse.getErrorMessage() != null ? "because" + sendResponse.getErrorMessage() : ""),
-                            null);
+                            sendResponse.getErrorMessage() != null ? "because" + sendResponse.getErrorMessage() : "");
+                    onError(updatedRegistration, errorMsg,
+                            new InvalidRequestException(sendResponse.getErrorMessage() != null ? errorMsg : ""));
```

Comment:
I read it again and I'm not sure it make sense to have ""Invalid Send Request, server returns %s %s"" in exception description description :thinking: 

Maybe we should rather go with something like : 
```java
   onError(updatedRegistration, String.format("Invalid Send Request, server returns %s %s", //
                            sendResponse.getCode().getName(), //
                            sendResponse.getErrorMessage() != null ? "because" + sendResponse.getErrorMessage() : ""),
                            new InvalidRequestException(sendResponse.getErrorMessage() != null ? sendResponse.getErrorMessage() : "unknown reason"));
```

@JaroslawLegierski  WDYT ?

Reply:
I think that is very good idea.